commons/src/main/java/com/powsybl/commons/datasource/MultipleReadOnlyDataSource.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.datasource;
+
+import com.powsybl.commons.PowsyblException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class MultipleReadOnlyDataSource implements ReadOnlyDataSource {
+
+    private final List<ReadOnlyDataSource> dataSources;
+
+    public MultipleReadOnlyDataSource(ReadOnlyDataSource... dataSource) {
+        this(List.of(dataSource));
+    }
+
+    public MultipleReadOnlyDataSource(List<ReadOnlyDataSource> dataSources) {
+        this.dataSources = Objects.requireNonNull(dataSources);
+        if (dataSources.isEmpty()) {
+            throw new PowsyblException("Empty data source list");
+        }
+    }
+
+    @Override
+    public String getBaseName() {
+        return dataSources.get(0).getBaseName(); // this is just a convention
```

Comment:
Wouldn't it be a bit more understandable if we concatenate?

Reply:
It won't work in any cases if we concatenate. Base name is used for (prefix, ext) query and whatever the way we select it, it won't be really useful.